entries/git-workflow-guidelines/commits.md

```diff
@@ -0,0 +1,213 @@
+
+Note that you can add as many files and changes to the staging area as you want before you commit them. All changes in the staging area at the time of commit will be a part of the same commit. In this way, the staging area allows you to control which changes you want to include in a specific commit, so you can collect the changes connected to the same unit of work in one commit (we will return to what that means in the section [Atomic Commits](#atomic-commits) below).
+
+Commits are always created locally in the specific branch you are currently on. Therefore, it is important to check that you are on the right branch before you commit (check the [Branching](branching.md#branching-in-practice) post for how you check which branch you are currently on and how to change branch). After you have created your commits, they needs to be *pushed* to the remote repository (on GitHub or or any other version control service that uses Git) for anyone else to be able to see it.
+
+After you have pushed your commits, your work will be backed up on the remote storage. As your work progresses, you continuously commit and push changes to the branch you are working on until you are ready to ask for feedback through a pull request (more on pull requests and the review process in the [Pull Request](prs.md) post).
+
+## Atomic Commits
+
+Now you should have an understanding of that commits are, but how often should you commit, and how do you choose what to include in a commit?
+
+Ideally, you should commit often, based around isolated, complete changes, a practice called **atomic** commits. This means that each commit should document a *single, complete unit of work* that represents a specific idea (as described in the post [Make Atomic Git Commits](https://www.aleksandrhovhannisyan.com/blog/atomic-git-commits/#atomic-commits-and-the-single-responsibility-principle) and the [Git guides](https://github.com/git-guides/git-commit)). As a result, that one commit should do one - and only one - thing, which can be summed up in a short, simple sentence (i.e., in the commit message).
+
+An atomic commit can also be described as a commit of the smallest possible size, that can be undone without any unwanted side effects apart from what would be expected based on its commit message. A commit is not atomic, if undoing it removes other changes than described.
+
```

Comment:
Does "the smallest possible size" sound a bit like you should commit every little change individually?

Reply:
I think we should be more clear about commit vs push. The one commit should be one smaller but also complete change which makes the git reverse more easier.

Reply:
Good point. I'll add something about that :)